optax/_src/utils.py

```diff
@@ -23,28 +25,29 @@
 from optax._src import numerics
 
 
-def tile_second_to_last_dim(a):
+def tile_second_to_last_dim(a: chex.Array) -> chex.Array:
   ones = jnp.ones_like(a)
   a = jnp.expand_dims(a, axis=-1)
   return jnp.expand_dims(ones, axis=-2) * a
 
 
-def canonicalize_dtype(dtype):
+def canonicalize_dtype(dtype: Optional[jnp.dtype]) -> Optional[jnp.dtype]:
```

Comment:
I think we should make this `chex.ArrayDType` - I'm not sure whether `jnp.dtype` works if a user calls the function with e.g. just float. Also, that way we use chex wherever possible (consistent with `chex.Array`).